pkg/runtime/processor/components.go

```diff
@@ -44,7 +45,19 @@ func (p *Processor) Init(ctx context.Context, comp componentsapi.Component) erro
 		return errors.Join(err, p.compStore.DropPendingComponent())
 	}
 
-	return p.compStore.CommitPendingComponent()
+	if err := p.compStore.CommitPendingComponent(); err != nil {
+		return fmt.Errorf("error committing component: %w", err)
+	}
+
+	if p.componentCb != nil {
+		if err := p.componentCb(registry.ComponentRegistry{
```

Comment:
This is not capturing init errors. I’m not sure if that’s intended.. we likely need to update the signature and wrap this init and close to pass that on.

Reply:
It also follows that we should close the component of callback fails.